components/bl-double-calendar-component/src/helpers/use-actions.js

```diff
@@ -1,18 +1,23 @@
-import { differenceInTime } from './utils';
+import { differenceInTime, normalizeFormat } from './utils';
+
+export function useActions(props) {
+  const { component, dateFormat, startDate, endDate, daysAmount, setStartDate, setEndDate } = props;
 
-export function useActions({ component, startDate, endDate, daysAmount, setStartDate, setEndDate }) {
   Object.assign(component, {
     getFromDate     : () => startDate,
-    setFromDate     : fromDate => setStartDate(new Date(fromDate)),
+    setFromDate     : fromDate => setStartDate(new Date(normalizeFormat(fromDate, dateFormat))),
     getToDate       : () => endDate,
-    setToDate       : toDate => setEndDate(new Date(toDate)),
+    setToDate       : toDate => setEndDate(new Date(normalizeFormat(toDate, dateFormat))),
     getFromAndToDate: () => ({ fromDate: startDate, toDate: endDate }),
     setFromAndToDate: (fromDate, toDate) => {
-      const diffInTime = differenceInTime(new Date(fromDate), new Date(toDate));
+      const normalizedFromDate = new Date(normalizeFormat(fromDate, dateFormat));
+      const normalizedToDate = new Date(normalizeFormat(toDate, dateFormat));
+
+      const diffInTime = differenceInTime(normalizedFromDate, normalizedToDate);
 
       if (diffInTime >= 0) {
-        setStartDate(new Date(fromDate));
-        setEndDate(new Date(toDate));
+        setStartDate(normalizedFromDate);
+        setEndDate(normalizedToDate);
```

Comment:
I assume `normalizedFromDate` and `normalizedToDate` could be passed as argument here, check it please.

Reply:
Dates that passed to this function came from user input, that's why we need to normalize it here. If I understood you correctly.